cache/remote.go

```diff
@@ -212,7 +212,7 @@ func (sr *immutableRef) getRemote(ctx context.Context, createIfNeeded bool, refC
 			}
 		}
 
-		if refCfg.Compression.Force {
+		if refCfg.Compression.Force || needsForceCompression(desc, refCfg.Compression.Type) {
```

Comment:
If the main blob (returned from `ref.ociDesc()`) is nydus, does this always trigger force compression every later compression variant requests? Maybe we should avoid this.